heka/sandbox/filters/firefox_plugin_hangs.lua

```diff
@@ -0,0 +1,129 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Firefox Plugin Hangs
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [FirefoxPluginHangs]
+    type = "SandboxFilter"
+    filename = "lua_filters/firefox_plugin_hangs.lua"
+    message_matcher = "Logger == 'fx' && Type == 'executive_summary'"
+    ticker_interval = 60
+    preserve_data = true
+--]]
+
+
+require "circular_buffer"
+require "string"
+local fx = require "fx"
+
+channels = {}
+
+local BUILD_IDS  = 6
+local DAYS       = 180
+local SEC_IN_DAY = 60 * 60 * 24
+
+local function set_header(cb, col, id)
+    cb:set_header(col, id, "hangs/hour", "none")
+end
+
+local channel_cnt = fx.get_channel_count()
+for i=1, channel_cnt do
+    local hph = circular_buffer.new(DAYS, BUILD_IDS, SEC_IN_DAY)   -- hangs per hour
+    local hours = circular_buffer.new(DAYS, BUILD_IDS, SEC_IN_DAY) -- total hours of use
+    local hangs = circular_buffer.new(DAYS, BUILD_IDS, SEC_IN_DAY) -- total number of hangs
+    channels[fx.get_channel_name(i-1)] = {hph = hph, hours = hours, hangs = hangs, ids = {}}
+    for j=1, BUILD_IDS do
+        set_header(hph, j, "unknown")
+    end
+end
+
+local function find_build_id(channel, id)
```

Comment:
Does this just keep the first 6 (or `BUILD_IDS`) builds we see for each channel for each day?  or for the entire reporting period? or the most recently seen 6?